flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/operators/windowing/ProcessingTimeoutTriggerTest.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.runtime.operators.windowing;
+
+import org.apache.flink.streaming.api.windowing.triggers.CountTrigger;
+import org.apache.flink.streaming.api.windowing.triggers.ProcessingTimeoutTrigger;
+import org.apache.flink.streaming.api.windowing.triggers.TriggerResult;
+import org.apache.flink.streaming.api.windowing.windows.TimeWindow;
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
+
+
+import org.junit.Test;
+
+import static org.apache.flink.core.testutils.CommonTestUtils.assertThrows;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Tests for {@link ProcessingTimeoutTrigger}.
+ */
+public class ProcessingTimeoutTriggerTest {
+
+	@Test
+	public void testWindowFireWithoutContinualInterval() throws Exception {
```

Comment:
Please add smaller isolated tests for the individual situations. Some of the things we need to test are:
 - state clearing, timer clearing
 - whether `shouldClearAtTimeout` and `resetTimerOnNewRecord` work
 - whether we correctly forward `PURGE` from the nested timer

Reply:
I am starting working on new tests, but i have a question.
it seems like `ctx.getCurrentProcessingTime()` in tests always return Long.MIN_VALUE, deep down to the test framework code, seem it can be changed by calling `testHarness.advanceProcessingTime` and it will set the currentProcessingTime, the only problem that it is throw exception when there is no firing.
So is there another way to change processingTime that `ctx.getCurrentProcessingTime` will return my value? (except of Mock it). or the only way is to call the `testHarness.advanceProcessingTime` method and catch/assert the exception.

Reply:
So eventually, I used the advanceProcessingTime, created multiple new tests the cover all scenario above.